### ISSUE TITLE
FCN training tutorial with nan

### ISSUE THREAD

**Issue by BrunoKrinski** [dmlc/gluon-cv#1086]
Hi, I'm trying to run a FCN model in my segmentation problem. I tried to follow the "4. Train FCN on Pascal VOC Dataset" tutorial without any modification and when I ran the code to train with several iterations, the loss starts to be 'nan' after a few iterations. Does someone know what could be the problem? If it is the tutorial that is with an error or perhaps a configuration that is missing? I'm grateful if someone could help me.

**Comment by bryanyzhu** [comment]
Could you provide more details? Are you running the tutorial on VOC dataset or on your own segmentation dataset? Usually `NaN` in the first few iterations is because of large learning rate or having some bad data. 

Try to follow the tutorial on VOC dataset, if you still encounter `NaN`, please report back with training log and your environment setup (ubuntu version, cuda version, conda or not, gluoncv version, etc.). Thank you.

**Comment by BrunoKrinski** [comment]
Initially, I tried to run in my own dataset and had NaN in the loss. Then, I tried to run the tutorial without any modification (in the voc dataset). I copied the tutorial to a file, commented the last lines to run more iterations, and continued the NaN. 
`# just demo for 2 iters`
 `#if i > 1:`
` #    print('Terminated for this demo...')`
 `#     break`

**Comment by bryanyzhu** [comment]
Got it, let me test it on my end.

**Comment by bryanyzhu** [comment]
The NaN is because the `batch_size` is too small and learning rate is too high. Our tutorial is just to show the workflow of training, not the actual training. 

If you start the actual training with command shown in the beginning of the tutorial, there is no NaN problem. 

```
CUDA_VISIBLE_DEVICES=0,1,2,3 python ./scripts/segmentation/train.py --dataset pascal_aug --model fcn --backbone resnet50 --lr 0.001 --checkname mycheckpoint
```

FYI, semantic segmentation is very sensitive to batch size.

**Comment by BrunoKrinski** [comment]
I tried to reduce the learning rate and use batch_size 4. It seems to be working now. I will try to adapt to my own datasets. Thanks.

**Comment by bryanyzhu** [comment]
Glad to know it is working, I will close the issue for now. Feel free to reopen it or open a new issue if you have problems on adapting to your own dataset.

**Comment by Tingting-Lu** [comment]
hi @BrunoKrinski 
Could you please tell me how to save the model after training? I try to use model.save_parameters, But it doesn't work.
Thanks

**Comment by joshuachang2311** [comment]
> hi @BrunoKrinski
> Could you please tell me how to save the model after training? I try to use model.save_parameters, But it doesn't work.
> Thanks

#1357

**Comment by joshuachang2311** [comment]
@Tingting-Lu #1357
